test/utils/HaiTest.t.sol

```diff
@@ -133,6 +133,14 @@ contract OverflowChecker {
 
 abstract contract HaiTest is DSTestPlus, OverflowChecker {
   modifier mockAsContract(address _address) {
+    // Foundry fuzzer sometimes gives us the next deployment address
+    // this results in very unexpected reverts as any contract deploy will revert
+    // we check here to make sure its not the next deployment address for the (pranked) msg.sender
+    (, address _msgSender,) = vm.readCallers();
+    address _nextDeploymentAddr = computeCreateAddress(address(_msgSender), vm.getNonce(_msgSender));
+
+    vm.assume(_address != _nextDeploymentAddr);
```

Comment:
Hmm. The mocked contract address may still collide with the address of a possible second deployment. Besides, the reversions may still happen upon deploying from currently inactive msg senders. 🤔

Reply:
can u run a `yarn test:unit:deep` on that test and see if it collides? if it doesn't, i'm ok with it

Reply:
It does not collide. 🎉